client/src/legacy/CMSMain.EditForm.js

```diff
@@ -476,4 +477,43 @@ $.entwine('ss', function($){
 	if ($('.cms-edit-form.CMSPageSettingsController input[name="ParentType"]:checked').attr('id') == 'Form_EditForm_ParentType_root') {
 		$('.cms-edit-form.CMSPageSettingsController #Form_EditForm_ParentID_Holder').hide(); //quick hide on first run
 	}
+
+  var confirmed = false;
+  /**
+   * Warn the user not to remove the homepage
+   */
+  $('.cms-edit-form .btn-toolbar #Form_EditForm_action_unpublish.homepage-warning,' +
+	'.cms-edit-form .btn-toolbar #Form_EditForm_action_archive.homepage-warning,' +
+	'#Form_EditForm_URLSegment_Holder.homepage-warning .btn.update').entwine({
+    onclick: async function (e) {
+      if (confirmed) {
+        return this._super(e);
+      }
+      e.stopPropagation();
+
+      var message = i18n._t(
+        'CMS.RemoveHomePageWarningMessage',
+        'Warning: This page is the home page. ' +
+        'By changing the URL segment visitors will not be able to view it.'
+	  );
+
+      if (await reactConfirm(message, {
+        title: i18n._t(
+          'CMS.RemoveHomePageWarningTitle',
+          'Remove your home page?'
+        ),
+	    confirmLabel: i18n._t(
+          'CMS.RemoveHomePageWarningLabel',
+          'Remove'
+        ),
+        confirmColor: 'danger'
+	  })) {
```

Comment:
Your indentation is still pretty whack - have you done a thorough self review on this PR?

Reply:
Should be fixed now. :)